src/components/image-content-block/image-content-block.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { Trans } from 'react-i18next';
+import styled, { css } from 'styled-components';
+import { up } from '../breakpoint/breakpoint';
+import { theme } from '../layout/theme';
+import { TextStyles } from '../typography/typography-v2';
+
+const mapTextAlignToFlex = (textAlign: 'right' | 'left' | 'bottom') => {
+  if (textAlign === 'right') {
+    return 'row';
+  } else if (textAlign === 'left') {
+    return 'row-reverse';
+  } else {
+    return 'column';
+  }
+};
+
+interface TextAlignProps {
+  textAlign: 'right' | 'left' | 'bottom';
+}
+
+const BlockWrapper = styled.div<TextAlignProps>`
+  display: inline-flex;
+  flex-direction: column;
+
+  gap: 12px 24px;
+
+  width: 100%;
+
+  ${up('sm')} {
+    flex-direction: ${({ textAlign }) => mapTextAlignToFlex(textAlign)};
+  }
+`;
+
+const ImageWrapper = styled.div`
+  display: inline;
+  position relative;
+`;
+
+const AttributionContainer = styled.div`
+  position: absolute;
+  bottom: 0;
+  right: 0;
+  background: rgba(0, 0, 0, 0.2);
+  display: block;
+
+  padding: 6px 8px;
+  margin-bottom: 5px;
+
+  font-size: 10px;
+  line-height: 110%;
+  color: #ffffff;
+`;
+
+const StyledLink = styled.a`
+  color: #ffffff;
+`;
+
+const TextWrapper = styled.div<TextAlignProps>`
+  ${TextStyles.textXS}
+
+  max-width: 100%;
+  color: ${theme.palette.text.default};
+
+  ${up('sm')} {
+    ${({ textAlign }) =>
+      textAlign === 'bottom' &&
+      css`
+        ${TextStyles.textXS}
+      `}
+
+    ${({ textAlign }) =>
+      textAlign !== 'bottom' &&
+      css`
+        ${TextStyles.textS}
+        max-width: 228px;
+      `}
+  }
+`;
+
+interface ImageContentBlockProps {
+  children: React.ReactNode;
+  textAlign: 'right' | 'left' | 'bottom';
```

Comment:
I really don't like the name `textAlign` because you are changing the layout and not the text alignment. You are creating a false expectation from the API. I would name this `textPlacement` or maybe you have some other alternative.

Reply:
👍